File: videohub/backend/app/services/permissions.py
"""功能点权限码:细粒度只读授权的单一事实来源。

与前端 src/utils/permissions.js 同步维护;新增可控功能点 = 在此加码并接线端点,
users 表无需迁移。授权语义为「只读可见」(设计:
docs/superpowers/specs/2026-09-15-permission-system-design.md)——写操作仍一律 require_admin。
"""
PERMISSION_CODES: frozenset[str] = frozenset({"dashboard", "routes", "users", "access"})

File: videohub/frontend/src/utils/permissions.js
// 功能点权限码:与后端 app/services/permissions.py 同步维护(单一事实来源在后端)。
// 授权语义为只读可见;写操作入口一律以 is_admin 判定显隐。
export const PERMISSIONS = [
  { code: "dashboard", label: "BI 大屏" },
  { code: "routes", label: "线路配置(只读)" },
  { code: "users", label: "用户管理(只读)" },
  { code: "access", label: "访问记录(只读)" },
]
